chainer/functions/pooling_2d.py

```diff
@@ -352,3 +355,105 @@ def average_pooling_2d(x, ksize, stride=None, pad=0, use_cudnn=True):
 
     """
     return AveragePooling2D(ksize, stride, pad, False, use_cudnn)(x)
+
+
+class SpatialPyramidPooling2D(function.Function):
+
+    """Spatial pyramid pooling over a set of 2d planes."""
+
+    def __init__(self, x, pyramid_height, pooling_class, use_cudnn=True):
```

Comment:
`__init__`  uses only the shape of `x` and do not use its data. So second argument should be `shape` instead of `x` .